include/taskolib/GitRepository.h

```diff
@@ -0,0 +1,175 @@
+/**
+ * \file   GitRepository.h
+ * \author Sven-Jannik Wöhnert
+ * \date   Created on March 20, 2023
+ * \brief  Wrapper for C-Package libgit2
+ *
+ * \copyright Copyright 2023 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+
+#include <git2.h>
+#include <string>
+#include <filesystem>
+#include "taskolib/GitObjectWrapper.h"
+#include <vector>
+
+namespace task {
+
+/**
+ * Struct to express the git status for one file
+ * \param path_name:  relative path to file.
+ *                    If the path changed this value will
+ *                    have the shape "OLD_NAME -> NEW_NAME"
+ * \param handling:   Handling status [unchanged, unstaged, staged, untracked, ignored]
+ * \param changes:    Change status [new file, deleted, renamed, typechanged, modified, unchanged, ignored, untracked]
+ */
+typedef struct
+{
+    std::string path_name;
+    std::string handling;
+    std::string changes;
+} filestatus;
```

Comment:
In C++, there is no need for the `typedef`:
```
struct FileStatus
{ 
...
}
```
According to the style guide this is a type, so it needs to be `InCamelCase`.
Finally, Doxygen `\param` only works for function parameters. You can add inline comments like this, though:
```
struct FileStatus
{
    std::string path_name; ///< Relative path to the file 
    std::string handling; ///< Handling status ...
    std::string changes; ///< Bla...
} filestatus;
```